src/editor/shadow-taxonomies/child-listings.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * UI for managing child listings and posts.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import {
+	Button,
+	ExternalLink,
+	Modal,
+	Notice,
+	PanelRow,
+	ToggleControl,
+} from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { useEffect, useState } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * External dependencies
+ */
+import { AutocompleteWithSuggestions } from 'newspack-components';
+
+/**
+ * Internal dependencies
+ */
+import { getPostTypeLabel, getTaxonomyForPostType, isListing } from '../utils';
+import './style.scss';
+
+const ChildListingsComponent = ( { hideChildren, postId, updateMetaValue } ) => {
+	const [ modalVisible, setModalVisible ] = useState( false );
+	const [ isUpdating, setIsUpdating ] = useState( false );
+	const [ initialChildPosts, setInitialChildPosts ] = useState( [] );
+	const [ childPosts, setChildPosts ] = useState( [] );
+	const [ selectedPostType, setSelectedPostType ] = useState( null );
+	const [ message, setMessage ] = useState( null );
+	const postType = window?.newspack_listings_data.post_type;
+	const listingPostTypes = window?.newspack_listings_data.post_types || {};
+	const taxonomyForPostType = getTaxonomyForPostType();
+	const childPostTypes = taxonomyForPostType?.post_types || [];
+	const filteredChildPostTypes = childPostTypes.filter(
+		childPostType => childPostType !== postType
+	);
+	const canHaveChildren = 0 < filteredChildPostTypes.length;
+
+	// Fetch suggestions for suggestions list on component mount.
+	useEffect(() => {
+		setMessage( null );
+		apiFetch( {
+			path: addQueryArgs( '/newspack-listings/v1/children', {
+				per_page: 100,
+				post_id: postId,
+			} ),
+		} )
+			.then( response => {
+				if ( response ) {
+					const mappedResponse = response.map( post => ( {
+						value: post.value,
+						label: post.label,
+						postType: post.post_type,
+					} ) );
+					setChildPosts( mappedResponse );
+					setInitialChildPosts( mappedResponse );
+				}
+			} )
+			.catch( e => {
+				setMessage( {
+					status: 'error',
+					children: e.message || __( 'Error fetching suggestions.', 'newspack-listings' ),
+					isDismissible: false,
+				} );
+			} );
+	}, []);
+
+	// Bail early if the post type can't have child listings.
+	if ( ! canHaveChildren || ! isListing() ) {
+		return null;
```

Comment:
This bailing happens after the initial fetch, which is then wasteful. The fetching should not happen if the component will not render anything (e.g. when opening the editor on an unsupported post type).

Reply:
Fixed in 1902d9d.